sys/net/network_layer/ng_sixlowpan/ng_sixlowpan.c

```diff
@@ -0,0 +1,249 @@
+
+    if (sixlowpan == NULL) {
+        DEBUG("6lo: no space left in packet buffer\n");
+        ng_pktbuf_release(pkt);
+        return;
+    }
+
+    sixlowpan->next = ipv6;
+    pkt->next = sixlowpan;
+    disp = sixlowpan->data;
+    disp[0] = NG_SIXLOWPAN_UNCOMPRESSED;
+    payload_len++;
+
+    iface = ng_sixlowpan_netif_get(hdr->if_pid);
+
+    if (iface == NULL) {
+        if (ng_netapi_get(hdr->if_pid, NETCONF_OPT_MAX_PACKET_SIZE,
+                          0, &max_frag_size, sizeof(max_frag_size)) < 0) {
+            /* if error we assume it works */
+            DEBUG("6lo: can not get max packet size from interface %"
+                  PRIkernel_pid "\n", hdr->if_pid);
+            max_frag_size = UINT16_MAX;
+        }
+
+        ng_sixlowpan_netif_add(hdr->if_pid, max_frag_size);
+    }
+    else {
+        max_frag_size = iface->max_frag_size;
+    }
+
+    DEBUG("6lo: max_frag_size = %" PRIu16 " for interface %"
+          PRIkernel_pid "\n", max_frag_size, hdr->if_pid);
+
+    /* IP should not send anything here if it is not a 6LoWPAN interface,
+     * so we don't need to check for NULL pointers */
+    if (payload_len <= max_frag_size) {
+        DEBUG("6lo: Send SND command for %p to %" PRIu16 "\n",
+              (void *)pkt, hdr->if_pid);
+        ng_netapi_send(hdr->if_pid, pkt);
+
+        return;
+    }
+    DEBUG("6lo: packet too big (%u> %" PRIu16 ")\n",
+          (unsigned int)payload_len, max_frag_size);
+}
+
+static void *_sixlowpan_thread(void *args)
+{
+    msg_t msg, reply, msg_q[NG_SIXLOWPAN_MSG_QUEUE_SIZE];
+    ng_netreg_entry_t me_reg;
+
+    (void)args;
+    msg_init_queue(msg_q, NG_SIXLOWPAN_MSG_QUEUE_SIZE);
+
+    me_reg.demux_ctx = NG_NETREG_DEMUX_CTX_ALL;
+    me_reg.pid = thread_getpid();
+
+    /* register interest in all 6LoWPAN packets */
+    ng_netreg_register(NG_NETTYPE_SIXLOWPAN, &me_reg);
+
+    /* preinitialize ACK */
+    reply.type = NG_NETAPI_MSG_TYPE_ACK;
+
+    /* start event loop */
+    while (1) {
+        DEBUG("6lo: waiting for incoming message.\n");
+        msg_receive(&msg);
+
+        switch (msg.type) {
+            case NG_NETAPI_MSG_TYPE_RCV:
+                DEBUG("6lo: NG_NETDEV_MSG_TYPE_RCV received\n");
+                _handle_receive((ng_pktsnip_t *)msg.content.ptr);
+                break;
+
+            case NG_NETAPI_MSG_TYPE_SND:
+                DEBUG("6lo: NG_NETDEV_MSG_TYPE_SND received\n");
+                _handle_send((ng_pktsnip_t *)msg.content.ptr);
+                break;
+
+            case NG_NETAPI_MSG_TYPE_GET:
+            case NG_NETAPI_MSG_TYPE_SET:
+                DEBUG("6lo: reply to unsupported get/set\n");
+                reply.content.value = -ENOTSUP;
+                msg_reply(&msg, &reply);
+                break;
+
+            default:
+                DEBUG("6lo: operation not supported\n");
+                break;
+        }
+    }
+
+    return NULL;
+}
+
+kernel_pid_t ng_sixlowpan_init(void)
```

Comment:
Just a matter of taste, but in IPv6 you declared the public functions first. Being consistent with that eases reviewing and getting an overview.